cpp/include/cudf/detail/iterator.cuh

```diff
@@ -254,8 +277,7 @@ auto inline make_scalar_iterator(scalar const& scalar_value)
 {
   CUDF_EXPECTS(data_type(type_to_id<Element>()) == scalar_value.type(), "the data type mismatch");
   CUDF_EXPECTS(scalar_value.is_valid(), "the scalar value must be valid");
-  return thrust::make_transform_iterator(thrust::make_constant_iterator<size_type>(0),
-                                         scalar_value_accessor<Element>{scalar_value});
+  return make_counting_transform_iterator(0, scalar_value_accessor<Element>{scalar_value});
```

Comment:
I don't like replacing transform(constant) with transform(counting). These are not the same thing when you get to the assembly code. I don't see a need for this change, or a benefit from it.

Reply:
Oh yea, this is a total mistake. Great catch! I am surprised CI didn't break.